client/src/controllers/DrilldownController.test.js

```diff
@@ -0,0 +1,214 @@
+    application.register('w-dropdown', DropdownController);
+    application.start();
+
+    await jest.runAllTimersAsync();
+    controller = application.getControllerForElementAndIdentifier(
+      document.querySelector(`[data-controller="${identifier}"]`),
+      identifier,
+    );
+
+    actionElement = document.getElementById('action-element');
+
+    return [
+      ...document.querySelectorAll(`[data-controller~="${identifier}"]`),
+    ].map((element) =>
+      application.getControllerForElementAndIdentifier(element, identifier),
+    );
+  };
+
+  beforeEach(async () => {
+    await setup();
+  });
+
+  afterEach(() => {
+    application?.stop && application.stop();
+    errors = [];
+    eventNames.forEach((name) => {
+      events[name] = [];
+    });
+  });
+
+  describe('DrilldownController open/close actions', () => {
+    it('opens the submenu and sets active submenu value when toggle is clicked', async () => {
+      const toggleButton = document.querySelector(
+        '[data-w-drilldown-target="toggle"]',
+      );
+      const submenu = document.querySelector('#drilldown-field-0');
+
+      toggleButton.click();
+      await jest.runAllTimersAsync();
+
+      expect(controller.activeSubmenuValue).toBe('drilldown-field-0');
+      expect(toggleButton.getAttribute('aria-expanded')).toBe('true');
+      expect(submenu.hidden).toBeFalsy();
+    });
+
+    it('closes the submenu and clears active submenu value when close action is triggered', () => {
+      const toggleButton = document.querySelector(
+        '[data-w-drilldown-target="toggle"]',
+      );
+      toggleButton.click();
+
+      controller.close();
+
+      const submenu = document.querySelector('#drilldown-field-0');
+      expect(controller.activeSubmenuValue).toBe('');
+      expect(toggleButton.getAttribute('aria-expanded')).toBe('false');
+      expect(submenu.hidden).toBeTruthy();
+    });
+  });
+
+  describe('DrilldownController delayedClose method', () => {
+    it('should call close after 200ms delay', () => {
+      jest.spyOn(controller, 'close');
+      controller.delayedClose();
+      expect(controller.close).not.toHaveBeenCalled();
+      jest.advanceTimersByTime(200);
+      expect(controller.close).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('DrilldownController where the Outlets are connected/disconnected', () => {
+    it('should add event listener on outlet connection', () => {
+      controller.open = jest.fn();
+      controller.wActionOutletConnected(null, actionElement);
+      actionElement.click();
+      expect(controller.open).toHaveBeenCalled();
+    });
+
+    it('should remove event listener on outlet disconnection', () => {
+      controller.open = jest.fn();
+      controller.wActionOutletConnected(null, actionElement);
+      controller.wActionOutletDisconnected(null, actionElement);
+      actionElement.click();
+      expect(controller.open).not.toHaveBeenCalled();
```

Comment:
This is a tricky one, I would recommend you simply delete the DOM elements (the action outlets), run all timers async (await this) and then try to check for events you can listen to.

This, as written, is not really testing the behaviour but rather 'digging into' the implementation, which is best to avoid.